src/test/ui/rfc-2632-const-trait-impl/specializing-constness-2.rs

```diff
@@ -17,15 +18,14 @@ impl<T: Default> A for T {
     }
 }
 
-impl<T: Default + ~const Sup> const A for T {
+impl<T: Default + ~const Sup> const A for T { //~ ERROR: conflicting impl
     fn a() -> u32 {
         3
```

Comment:
This error should not occur. Please add a separate test to ensure that this passes compilation without the `generic` function.